frontend/src/components/MeetingRoom/CaptionsButton/CaptionsButton.tsx

```diff
@@ -0,0 +1,141 @@
+import { ClosedCaption, ClosedCaptionDisabled } from '@mui/icons-material';
+import { Tooltip } from '@mui/material';
+import { ReactElement, useState } from 'react';
+import useRoomName from '../../../hooks/useRoomName';
+import ToolbarButton from '../ToolbarButton';
+import { enableCaptions } from '../../../api/captions';
+import { SubscriberWrapper } from '../../../types/session';
+import CaptionsBox from './CaptionsBox';
+import useSessionContext from '../../../hooks/useSessionContext';
+
+export type CaptionsButtonProps = {
+  isOverflowButton?: boolean;
+  handleClick?: () => void;
+  subscriberWrappers: SubscriberWrapper[];
+};
+
+/**
+ * CaptionsButton Component
+ *
+ * Displays a button and handles the captioning functionality.
+ * @param {CaptionsButtonProps} props - the props for the component
+ *  @property {boolean} isOverflowButton - (optional) whether the button is in the ToolbarOverflowMenu
+ *  @property {(event?: MouseEvent | TouchEvent) => void} handleClick - (optional) click handler that closes the overflow menu in small viewports.
+ * @returns {ReactElement} - The CaptionsButton component.
+ */
+const CaptionsButton = ({
+  isOverflowButton = false,
+  handleClick,
+  subscriberWrappers,
+}: CaptionsButtonProps): ReactElement => {
+  const { currentCaptionsIdRef, ownCaptions, vonageVideoClient } = useSessionContext();
+  const roomName = useRoomName();
+  const [captionsId, setCaptionsId] = useState<string>('');
+  const [isCaptionsEnabled, setIsCaptionsEnabled] = useState<boolean>(false);
+  const title = isCaptionsEnabled ? 'Disable captions' : 'Enable captions';
+
+  const handleClose = () => {
+    // If the CaptionsButton is in the ToolbarOverflowMenu, we close the modal and the menu
+    if (isOverflowButton && handleClick) {
+      // Close the menu immediately to improve mobile UX
+      handleClick();
+    }
+  };
+
+  const handleCaptions = async (action: 'enable' | 'disable') => {
+    if (action === 'enable') {
+      if (currentCaptionsIdRef?.current) {
+        setIsCaptionsEnabled(true);
+        setCaptionsId(currentCaptionsIdRef.current);
+
+        // we need to signal to other users that we are joining the captions
```

Comment:
Could we rephrase this? It makes more sense to say we're enabling captions publishing and disabling captions publishing. Or maybe `unpublish` and `publish`?